packages/lit-ssr/src/test/lit/render-lit_test.ts

```diff
@@ -204,6 +204,16 @@ test('element with property', async (t: Test) => {
   );
 });
 
+test('element with `willUpdate`', async (t: Test) => {
```

Comment:
It would also be good to add an `integration` test to test `willUpdate` interaction with `hydration` and that the use cases work as expected. `LitElement` integration tests are here: https://github.com/Polymer/lit-html/blob/lit-next/packages/lit-ssr/src/test/integration/tests/basic.ts#L3972

For debugging, use `npm run test:integration -- --debug` or `Test: Integration (debug) (lit-ssr)` in the `Run` tab in VSCode:
![image](https://user-images.githubusercontent.com/1354932/105115820-63b33480-5a7e-11eb-854c-253ead57ae27.png)

Reply:
Done.